test/e2e/cert_manager_test.go

```diff
@@ -0,0 +1,133 @@
+const cmCertificatealidity = 24 * time.Hour
+
+func TestCertManagerProvider(t *testing.T) {
+	feature := features.New("Cert-Manager Certificate").WithLabel("app", "cert-manager")
+
+	cmIssuer := &certv1.ClusterIssuer{
+		TypeMeta: metav1.TypeMeta{
+			Kind: cmIssuerType,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cmIssuerName,
+		},
+		Spec: certv1.IssuerSpec{
+			IssuerConfig: certv1.IssuerConfig{
+				SelfSigned: &certv1.SelfSignedIssuer{},
+			}},
+	}
+
+	cmConfig := &interfaces.Config{
+		CommonName:       cmCertificateName,
+		ValidityDuration: cmCertificatealidity,
+		ExtraConfig: map[string]any{
+			"issuerName": cmIssuerName,
+			"issuerKind": cmIssuerType,
+		},
+	}
+
+	feature.Setup(
+		func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			client := cfg.Client()
+			_ = appsv1.AddToScheme(client.Resources().GetScheme())
+			_ = corev1.AddToScheme(client.Resources().GetScheme())
+			_ = certv1.AddToScheme(client.Resources().GetScheme())
+			_ = apiextensionsV1.AddToScheme(client.Resources().GetScheme())
+
+			var issuer certv1.ClusterIssuer
+			if err := client.Resources().Get(ctx, cmIssuerName, cmCertificateNamespace, &issuer); err != nil {
+				if k8serrors.IsNotFound(err) {
+					t.Logf("No cert-manager selfsigned issuer found, creating clusterIssuer: %s", cmIssuerName)
+					// Create cert-manager issuer
+					if err := client.Resources().Create(ctx, cmIssuer); err != nil {
+						t.Fatalf("unable to create cert-manager selfsigned issuer: %s", err)
+					}
+				} else {
+					t.Fatalf("Failed to get cert-manager selfsigned issuer: %s", cmIssuerName)
+				}
+				t.Logf("Created clusterIssuer: %s", cmIssuerName)
+			}
+			return ctx
+		})
+
+	feature.Assess("Ensure certificate",
+		func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			client := cfg.Client()
+			cmProvider := cert_manager.New(client.Resources().GetControllerRuntimeClient())
+			err := cmProvider.EnsureCertificateSecret(ctx, cmCertificateName, cmCertificateNamespace, cmConfig)
+			if err != nil {
+				t.Fatalf("Cert-Manager Certificate could not be created: %v", err)
+			}
+			return ctx
+		})
+
+	feature.Assess("Validate certificate secret",
+		func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			client := cfg.Client()
+			cmProvider := cert_manager.New(client.Resources().GetControllerRuntimeClient())
+			err := cmProvider.ValidateCertificateSecret(ctx, cmCertificateName, cmCertificateNamespace, cmConfig)
+			if err != nil {
+				t.Fatalf("Failed to validate Cert-Manager Certificate secret: %v", err)
+			}
+			return ctx
+		})
+
+	feature.Assess("Get certificate config",
+		func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			client := cfg.Client()
+			cmProvider := cert_manager.New(client.Resources().GetControllerRuntimeClient())
+			_, err := cmProvider.GetCertificateConfig(ctx, cmCertificateName, cmCertificateNamespace)
```

Comment:
Can you verify the returned config match the original config that  is used to call EnsureCertificateSecret?